kolibri/core/content/utils/sqlalchemybridge.py

```diff
@@ -331,3 +335,30 @@ def end(self):
         self.session.close()
         if self.connection:
             self.connection.close()
+
+
+def filter_by_uuids(field, ids, validate=True):
+    return _by_uuids(field, ids, validate, True)
+
+
+def exclude_by_uuids(field, ids, validate=True):
+    return _by_uuids(field, ids, validate, False)
+
+
+def _by_uuids(field, ids, validate, include):
+    ids_list = list(ids)
+    query = "IN (" if include else "NOT IN ("
+    for (idx, identifier) in enumerate(ids_list):
+        if validate:
+            try:
+                UUID(identifier, version=4)
+            except (TypeError, ValueError):
+                # the value is not a valid hex code for a UUID, so we don't return any results
+                return UnaryExpression(field, modifier=operators.custom_op(query + ")"))
+        # wrap the uuids in string quotations
+        if django_connection.vendor == "sqlite":
+            ids_list[idx] = "'{}'".format(identifier)
+        elif django_connection.vendor == "postgresql":
```

Comment:
This is the part that we need to conditionalize, we only want to do this code block in the case that we are using postgresql *and* the field we are querying against is a UUIDField - because the LocalFile id is a CharField that's why it is breaking.

That said, at some point we should work out how to change the LocalFile id to a UUIDField instead.

Reply:
yeah, I saw it later, it's _theorically_ fixed, now if Travis did its work...